portend.py

```diff
@@ -111,17 +111,17 @@ def free(host, port, timeout=float('Inf')):
 
 	timer = timing.Timer(timeout)
 
-	while not timer.expired():
+	while True:
 		try:
 			# Expect a free port, so use a small timeout
 			Checker(timeout=0.1).assert_free(host, port)
 			return
 		except PortNotFree:
+			if timer.expired():
```

Comment:
The placing of this test ahead of the `time.sleep` call changes the semantics/timing of the expiration - now the sleep will be avoided on expiration instead of checking the expiration after the sleep. Perhaps that's a better behavior, but it wasn't mentioned in the bug report nor in this PR. Was that change in behavior intentional?

Reply:
I think this is the right behavior for the reason you mention, but I hadn't realized that it was different from the existing behavior; sorry for not mentioning it. Should it be kept as is?